### PR TITLE
hilla: add an article to explain how to cancel endpoint requests

### LINKED ISSUE #1477: Add an article for cancelling endpoint requests
Add an article under the “Advanced Topics” section for the feature of cancelling the endpoint requests.

- Mention since Hilla 1.2
- Describe the use case
- Give a full example with `AbortController`
  - Creating `AbortController`
  - Passing the `AbortSignal`
  - Calling `.abort()`
  - Mention aborting multiple endpoint requests from a single `AbortController` instance
- Mention the error handling implications, if any (the endpoint call rejects with `AbortError`)
- Update [the error handling article](https://hilla.dev/docs/application/error-handling) to mention `AbortError`, link to the new advanced article.

### PR DESCRIPTION
**Pull request by cromoteca**
This PR contains a new article about endpoint request cancellation and modifies the error handling page to mention `AbortError`s related to cancellation.

Fixes #1477

### CODE REVIEW

File: articles/advanced/cancelling-endpoint-requests.asciidoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+title: Cancelling endpoint requests
+description: TODO
```
Comment (tarekoraby) [inline articles/advanced/cancelling-endpoint-requests.asciidoc:3]: Let's remember to add the description before merging.

File: articles/advanced/cancelling-endpoint-requests.asciidoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+order: 200
+layout: page
+---
+
+[role="since:com.vaadin:vaadin@V1.2"]
+== Cancelling endpoint requests
+
+Sometimes it can be useful to programmatically cancel some requests, for example when they are taking too long
+or they are standing in queue despito no longer being needed. Hilla allows developers to cancel those
```
Comment (tarekoraby) [inline articles/advanced/cancelling-endpoint-requests.asciidoc:12]:
```suggestion
or they are standing in queue despite no longer being needed. Hilla allows developers to cancel those
```

**Review by tarekoraby — CHANGES_REQUESTED** [review]

File: articles/advanced/cancelling-endpoint-requests.asciidoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+[role="since:com.vaadin:vaadin@V1.2"]
+== Cancelling endpoint requests
+
+Sometimes it can be useful to programmatically cancel some requests, for example when they are taking too long
+or they are standing in queue despite no longer being needed. Hilla allows developers to cancel those
+request using standard https://developer.mozilla.org/en-US/docs/Web/API/AbortController[AbortController]s.
+
+All endpoint requests support an optional `init` parameter that can be used to pass an `AbortSignal` which will
+allow to cancel them.
```
Comment (dougjewsbury) [inline articles/advanced/cancelling-endpoint-requests.asciidoc:16]:
```suggestion
All endpoint requests support an optional `init` parameter that can be used to pass an `AbortSignal` which will allow them to be cancelled.
```

File: articles/application/error-handling.asciidoc [hunk 4]
```diff
@@ -79,3 +79,9 @@ Also see <<../pwa/cache-client-side-data#,this article>> on caching endpoint dat
 
 If the server replies with a response other than `200 OK`, and the string contained in the response is not valid JSON, an [classname]#EndpointResponseError# is raised.
 The exception contains the response text as message and the https://developer.mozilla.org/en-US/docs/Web/API/Response[Response] object in the `response` field.
+
+[role="since:com.vaadin:vaadin@V1.2"]
+== Cancelled requests
+
+When a request is cancelled programmatically, the endpoint call promise will resolve with an `AbortError`. Actually this only
+happens when you explicitly cancel request as explained in <<../advanced/cancelling-endpoint-requests#,this article>>.
```
Comment (dougjewsbury) [inline articles/application/error-handling.asciidoc:87]:
```suggestion
When a request is cancelled programmatically, the endpoint call promise will resolve with an `AbortError`.
This happens only when you explicitly cancel a request, as explained in <<../advanced/cancelling-endpoint-requests#,this article>>.
```

**Review by tarekoraby — APPROVED** [review]

**Comment by tarekoraby** [comment]
I think it would be good for someone for the Hilla team to also review this, if they haven't already.

File: articles/advanced/cancelling-endpoint-requests.asciidoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+layout: page
+---
+
+[role="since:com.vaadin:vaadin@V1.2"]
+== Cancelling Endpoint Requests
+
+Sometimes it can be useful to programmatically cancel some requests, for example when they are taking too long or are waiting in a queue despite no longer being needed.
+Hilla allows developers to cancel these requests using standard https://developer.mozilla.org/en-US/docs/Web/API/AbortController[AbortController]s.
+
+All endpoint requests support an optional `init` parameter that can be used to pass an `AbortSignal` which will allow them to be cancelled.
```
Comment (platosha) [inline articles/advanced/cancelling-endpoint-requests.asciidoc:14]: Should probably mention the new name, or remove the name entirely. It’s not much useful anyway.

```suggestion
All endpoint requests support an optional last object parameter, that can be used to pass an `AbortSignal`, which will allow them to be cancelled.
```

Comment (cromoteca) [reply articles/advanced/cancelling-endpoint-requests.asciidoc:14]: Good idea. The name can even change when using the multi-module generator. What if I skip it altogether?

```suggestion
All endpoint requests support an optional parameter that can be used to pass an `AbortSignal` which will allow them to be cancelled.
```

**Review by platosha — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -158,25 +158,6 @@
...
-  },
-  "license": "UNLICENSED",
-  "lint-staged": {
-    "frontend/**/*.{js,ts,json}": [
-      "prettier --write",
-      "npm run lint"
-    ]
-  },
-  "name": "no-name",
-  "scripts": {
```
Comment (tarekoraby) [inline package.json:174]: @cromoteca is this intended? Are those changes really needed for this PR?

**Review by tarekoraby — CHANGES_REQUESTED** [review]

**Review by tarekoraby — APPROVED** [review]

**Pull request merged**